### ISSUE TITLE
Global Styles: enable the ability to edit gradient & duotone palettes

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#36541]
## What problem does this address?

Currently, it's not possible to edit gradients and duotone palettes within Global Styles but this feels like a missed opportunity and something that users will want to do in the future.

## What is your proposed solution?

For now, would be great to show duotone palettes even if they can't be edited. Long term, would be great to have these both as options for users to engage with.

**Comment by noisysocks** [comment]
Is this really a WP 5.9 blocker? Could we not add it in WP 6.0? Or is there a forward compatibility concern I'm not aware of?

**Comment by annezazu** [comment]
I think it could be added for 6.0 personally :) @priethor do you have more context?

**Comment by apeatling** [comment]
+1 for 6.0

**Comment by priethor** [comment]
> @priethor do you have more context?

The most important part here is managing the duotone palettes globally; unless I'm missing something, as important as it is, it would be a nice-to-have rather than a blocker for 5.9.

**Comment by richtabor** [comment]
A couple ideas I'd like to float, cc @WordPress/gutenberg-design  

1. Should we add a "Filters" top-level item to the Styles sidebar? 

<img width="771" alt="CleanShot 2023-03-17 at 13 34 16" src="https://user-images.githubusercontent.com/1813435/225977586-ec854536-d71e-44dc-a0bf-080447998e82.png">



2. Perhaps there's a way to simplify the duotone edit/creation modal. I have a [prototype](https://www.figma.com/proto/oEm1ck3f5gfGYyOR1Fs71T/Gutenberg-Ideas?page-id=220%3A866&node-id=220-4546&viewport=-336%2C1367%2C0.5&scaling=min-zoom&starting-point-node-id=220%3A4546) to try it, basically pulling the color palette control and duotone control closer together.  

https://user-images.githubusercontent.com/1813435/225977752-c04f2e6b-d7bf-49bb-8ddd-4cc72af1575f.mp4

- The preview image is particular to a global styles instance, when you probably won't see the duotone filter applied when editing. 
- When you select to edit one of the duotone colors, the color palette control renders below. This is in contrast to how duotone is applied today, where selecting a color opens a custom color palette. And there's extra UI for applying a preset to the highlight/shadow. 
- The color palette control would work exactly like it does elsewhere—click on the color preview to open the custom color popover. 
- If something like this gets in, we should do the same for the duotone toolbar control on blocks.

**Comment by jasmussen** [comment]
> Should we add a "Filters" top-level item to the Styles sidebar?

Maybe, yes!

My current mental model for Global Styles items is that each drilldown should match a panel name in the inspector. There isn't one for filters, but #39452 actually suggests adding such a panel:

<img src="https://user-images.githubusercontent.com/1204802/158345857-6c555027-9726-4bbc-b1a1-d956df873726.png">

With a counterpart in global styles, this would then show the Filters → Duotone detail page: 

<img src="https://user-images.githubusercontent.com/1204802/158345426-439bc56b-506d-4dce-9435-1a4ac6399b60.png">

Whether that happens in a flyout, or in the inspector, no strong opinion.

**Comment by MaggieCabrera** [comment]
> A couple ideas I'd like to float, cc @WordPress/gutenberg-design
> 
>     1. Should we add a "Filters" top-level item to the Styles sidebar?
> 

This aligns with what I discussed with @ajlende and the direction the duotone filters are taking

> 
>     2. Perhaps there's a way to simplify the duotone edit/creation modal. I have a prototype to try it, basically pulling the color palette control and duotone control closer together.
> 


>     * The preview image is particular to a global styles instance, when you probably won't see the duotone filter applied when editing.

Should we use the preview component that we use on other parts of GS?

<img width="279" alt="Screenshot 2023-03-20 at 09 47 22" src="https://user-images.githubusercontent.com/3593343/226289758-1182a9b8-fff8-4168-a6b0-15a6801dba35.png">



>     * If something like this gets in, we should do the same for the duotone toolbar control on blocks.

There is some discussion about this [here](https://github.com/WordPress/gutenberg/issues/39452#issuecomment-1437402674)

**Comment by richtabor** [comment]
> My current mental model for Global Styles items is that each drilldown should match a panel name in the inspector. 

Do you think we'd have a top-level "Filters" menu item, which navigates right to "Duotone" or would there be an intermediate step for selecting "Duotone"? Or would Filters > Duotone be fine for now, and when/if we add more filter types, we add the intermediate step? 

> Should we use the preview component that we use on other parts of GS?

Yes, the BlockPreview would be fine. It needs tidying up for images though. 

> Whether that happens in a flyout, or in the inspector, no strong opinion.

I like the presentation of the Duotone preview in the Inspector. My concern though, is that the preview/edit controls are _above_ the presets, as you need to select a preset first—then go back above to edit the preview. 

Feels a bit backwards. 

![CleanShot 2023-03-20 at 10 49 01](https://user-images.githubusercontent.com/1813435/226377395-549a83f8-e178-465e-b92c-aa995407bf08.png)

My prototype enforces the notion of clicking on a preset to edit it, just like how colors and gradients function—though the Duotone + Color palette may be a bit funky together. 

https://user-images.githubusercontent.com/1813435/226379063-f9c37dbb-70a2-4d4c-b733-fe1e402fc633.mp4

**Comment by jasmussen** [comment]
> Or would Filters > Duotone be fine for now, and when/if we add more filter types, we add the intermediate step?

Seems fine to add intermediate steps when they become available and not before.

> I like the presentation of the Duotone preview in the Inspector. My concern though, is that the preview/edit controls are above the presets, as you need to select a preset first—then go back above to edit the preview.

Solid point. Edit: which ever gets implemented, let's just remember to also close the other issue! 🙏

**Comment by richtabor** [comment]
> My prototype enforces the notion of clicking on a preset to edit it, just like how colors and gradients function—though the Duotone + Color palette may be a bit funky together.

@jameskoster any thoughts on how this could be improved? Do you think it's a bit funky?

**Comment by MaggieCabrera** [comment]
I'm working on adding the ability to edit and create presets for duotone filters, and I'm using the colors panel as a reference.

The code is shared between the panel that defines the colors globally and the panel that defines it for a specific block or element. This makes really weird interactions since you are allowed to edit the colors of the theme palette within a block, but things never get really saved:

![ezgif-4-789b8a4934](https://user-images.githubusercontent.com/3593343/227164260-1325cb08-8637-40da-9b47-859f1e79a21f.gif)


I'm guessing this is not a desired interaction. Should the palette panel only show when we are editing on the global level instead of at the block? And mimic the same for duotone filters and only allow for preset creation/editing on the global level?

/cc @richtabor @jasmussen

**Comment by jasmussen** [comment]
> This makes really weird interactions since you are allowed to edit the colors of the theme palette within a block, but things never get really saved:

Yeah, I've noticed this a couple of times too. The palette should not be available inside the individual blocks, that detail page should be entirely scoped to be just about the block and nothing else IMO. @annezazu do you know if there's an issue for this? Happy to open otherwise.

**Comment by jameskoster** [comment]
> My prototype enforces the notion of clicking on a preset to edit it, just like how colors and gradients function—though the Duotone + Color palette may be a bit funky together.

@richtabor the UX works well imo, but I wonder if the initial view might be better if we preselected the first color. IE show folks this straight away:

<img width="258" alt="Screenshot 2023-03-23 at 14 49 33" src="https://user-images.githubusercontent.com/846565/227241809-e399ce1e-ec93-4fa5-bbc3-d41be08eec2e.png">


Rather than this:

<img width="228" alt="Screenshot 2023-03-23 at 14 49 37" src="https://user-images.githubusercontent.com/846565/227241869-b13643d4-5108-4070-a79b-a22d048b8a16.png">

Speaking of which, those little white-outline circles could use an active state.

Side note, it might be good to look at this in the block toolbar as well because this _is_ funky:

<img width="467" alt="Screenshot 2023-03-23 at 14 54 59" src="https://user-images.githubusercontent.com/846565/227243302-c0ac021b-d0c6-4a87-8cf0-3fe41ff8cbe3.png">

**Comment by MaggieCabrera** [comment]
> Side note, it might be good to look at this in the block toolbar as well because this _is_ funky:
> 


@jameskoster  Would it make sense to remove duotone from the toolbar altogether since we are also going to have the controls in the sidebar inspector? I've started working on that in https://github.com/WordPress/gutenberg/pull/48405

**Comment by jasmussen** [comment]
I would not remove duotone from the toolbar without a fairly good reason. The main purpose of the block toolbar is to provide convenient quick access to the most important tools. Duotone fits that bill, IMO.

**Comment by jameskoster** [comment]
Yeah I would leave it there, but I don't know if it should remain possible to edit the values, that's when things gets overwhelming. Worth further exploration in a separate issue, but I'd expect options in the toolbar to:

1. Select a preset, or 
2. Create a custom filter
 
The edit interface would only be visible when the latter is selected.

**Comment by jasmussen** [comment]
Agreed, we could vastly simplify what opens from the toolbar, if that helps. Even access to only presets seems fine, worst case we could have an "edit" link at the bottom.

**Comment by jordesign** [comment]
> I would not remove duotone from the toolbar without a fairly good reason.

+1 to this. This would be a pretty major 'moved the cheese' situation for all the users who have had plenty of time to get used to accessing it within the toolbar.

**Comment by annezazu** [comment]
👋🏼 Hey folks, my past self included. I am doing a [sweep of high priority issues](https://wordpress.slack.com/archives/C02QB2JS7/p1702996696603749) and have removed this label. This issue has both stalled and [isn't mapped out for the next WordPress release](https://make.wordpress.org/core/2023/12/07/roadmap-to-6-5/). If that changes, let's add the label back but, for now, this clearly isn't a high priority issue when looking across the project.
